src/core/qgslocalizeddatapathregistry.cpp

```diff
@@ -82,6 +94,8 @@ void QgsLocalizedDataPathRegistry::registerPath( const QString &path, int positi
 
 void QgsLocalizedDataPathRegistry::unregisterPath( const QString &path )
 {
+  QgsReadWriteLocker locker( mLock, QgsReadWriteLocker::Write );
+
   mPaths.removeAll( QDir( path ) );
   writeToSettings();
```

Comment:
This is a potential dead lock.

1. Acquires a write lock
2. Calls `writeToSettings()`
3. Calls `paths()`
4. Tries to acquire a read lock which fails because there's a write lock in place